### PR TITLE
APP-3330: Fixed regex in SymMessageParser

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
[APP-3330](https://perzoinc.atlassian.net/browse/APP-3330)

### Description
Fixed regex in SymMessageParser to avoid ReDOS. Added unit tests.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added

### CODE REVIEW

File: lib/SymMessageParser/index.js [hunk 1]
```diff
@@ -2,14 +2,13 @@ var SymBotAuth = require('../SymBotAuth')
...
 
-const regex = /(<div.+?>)(.*)(<\/div>)/g
+const regex = /^(<div.+?>)(.*)(<\/div>)$/g
 const regexTagStart = /<\w+>/g
 const regexTagEnd = /<\/\w+>/g
 
 const HASHTAG_TYPE = 'org.symphonyoss.taxonomy.hashtag'
 const CASHTAG_TYPE = 'org.symphonyoss.fin.security.id.ticker'
 const MENTION_TYPE = 'com.symphony.user.userId'
-const EntityRegex = 'id=\"\\d+\">(.*?)<'
```
Comment (symphony-elias) [inline lib/SymMessageParser/index.js:12]: This was not used in the code and is not documented.

File: lib/SymMessageParser/index.js [hunk 2]
```diff
@@ -2,14 +2,13 @@ var SymBotAuth = require('../SymBotAuth')
 
 var SymMessageParser = {}
 
-const regex = /(<div.+?>)(.*)(<\/div>)/g
+const regex = /^(<div.+?>)(.*)(<\/div>)$/g
```
Comment (symphony-youri) [inline lib/SymMessageParser/index.js:5]: could it be a problem if we have _whitespace_<div>.... ?

Comment (symphony-elias) [reply lib/SymMessageParser/index.js:5]: Indeed... Let me update that.

File: lib/SymMessageParser/index.js (hunk 1 above)
Comment (andreihincu) [reply lib/SymMessageParser/index.js:12]: hello @symphony-elias , unfortunately, this regex still has ambiguity.

**Review by andreihincu — DISMISSED** [review]
Much better and cleaner....Great job, @symphony-elias !

File: tests/SymMessageParser/index.test.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      'Hello wonderful world');
+  });
+
+  test('message with several levels of markups', () => {
+    assertMessageParsed(
+      '<div data-format="PresentationML" data-version="2.0">Hello <b>wonderful <c>and</c> beautiful</b> world</div>',
+      'Hello wonderful and beautiful world');
+  });
+
+  test('message with several levels of markups with spaces around div', () => {
```
Comment (symphony-youri) [inline tests/SymMessageParser/index.test.js:60]: maybe add a test with a nested div?

**Review by symphony-youri — DISMISSED** [review]
👍 good job on removing the regex 
image

**Review by symphony-youri — APPROVED** [review]

**Pull request merged**